### PR TITLE
umlet: 15.0.0 -> 15.1.0

### PR DESCRIPTION
**Pull request by oxzi**
###### Description of changes
Based on <https://www.umlet.com/changes.htm>:
- Fix of a pesky, zero-size relation bug
- log4j mitigation (not a problem per se, but had triggered warnings)
- Dark mode tweaks

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/umlet/default.nix [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+  version = "15.1.0";
 
   src = fetchurl {
-    url = "https://www.umlet.com/umlet_${major}_${minor}/umlet-standalone-${version}.zip";
-    sha256 = "sha256-gdvhqYGyrFuQhhrkF26wXb3TQLRCLm59/uSxTPmHdAE=";
+    # NOTE: The download URL breaks consistency - sometimes w/ patch versions
+    # and sometimes w/o. Furthermore, for 15.1.0 they moved everything to the
+    # new /download subfolder.
+    # As releases are very rarely, just modify it by hand..
+    url = "https://www.umlet.com/download/umlet_15_1/umlet-standalone-15.1.zip";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/umlet/default.nix:12]: Maybe we should upload things to web.archive.org?

Comment (SuperSandro2000) [reply pkgs/tools/misc/umlet/default.nix:12]: https://web.archive.org/web/20230420091733/https://www.umlet.com/download/umlet_15_1/umlet-standalone-15.1.zip

File: pkgs/tools/misc/umlet/default.nix [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+    sha256 = "sha256-M6oVWbOmPBTygS+TFkY9PWucFfYLD33suNUuWpFLMIo=";
```
Comment (Janik-Haag) [inline pkgs/tools/misc/umlet/default.nix:13]:
```suggestion
    hash = "sha256-M6oVWbOmPBTygS+TFkY9PWucFfYLD33suNUuWpFLMIo=";
```

**Review by Janik-Haag — COMMENTED** [review]
LGTM; just a small nitpick

File: pkgs/tools/misc/umlet/default.nix (hunk 2 above)
Comment (oxzi) [reply pkgs/tools/misc/umlet/default.nix]: Changed; thanks.

**Pull request merged**